wicket-pivot/src/main/java/ro/fortsoft/wicket/pivot/tree/TreeHelper.java

```diff
@@ -34,6 +34,25 @@
 		return leafs;
 	}
 	
+	/*Using recursion to optimize the method by 
+	 * reducing the number of object created*/
+	public static int getLeafsSize(Node root) {
```

Comment:
Just a curiosity. It's clear that this method is better then `getLeafs(node).size()`. Another possible implementation is similar with `getLeafs(node)` but with a variable `leafsSize` instead of `leafs`:

```java
public static List<Node> getLeafs(Node root) {
	List<Node> leafs = new ArrayList<Node>();
	Iterator<Node> it = new TreeIterator(root);
	while (it.hasNext()) {
		Node node = it.next();
		if (node.isLeaf()) {
			leafs.add(node);
		}
	}
		
	return leafs;
}
```

```java
public static List<Node> getLeafsSize(Node root) {
	int leafsSize = 0;
	Iterator<Node> it = new TreeIterator(root);
	while (it.hasNext()) {
		Node node = it.next();
		if (node.isLeaf()) {
			leafsSize++;
		}
	}
		
	return leafsSize;
}
```

In this mode I eliminate the recursive call.